packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -900,3 +900,13 @@ enum Brightness {
   /// For example, the color might be bright white, requiring black text.
   light,
 }
+
+/// Process a VM service log event into a string message.
+String processVmServiceMessage(vm_service.Event event) {
+  final String message = utf8.decode(base64.decode(event.bytes));
+  // Remove extra trailing newlines appended by the vm service.
+  if (message.endsWith('\n')) {
```

Comment:
Would https://api.dart.dev/stable/2.9.1/dart-core/String/trimRight.html work?

Reply:
In the event the message contents itself has trailing whitespace, we should still retain that